app/views/posts/index.html.erb

```diff
@@ -1,7 +1,105 @@
-<h1>All Posts</h1>
-<p><%= @placeholder_text %></p>
-<ul>
-  <% @posts.each do |post| %>
-    <li><%= link_to post.title, post_path(post) %></li>
-  <% end %>
-</ul>
+<!DOCTYPE html>
+<html>
+<head>
+  <title>All Posts</title>
+  <style>
+    * {
+      box-sizing: border-box;
+      padding: 0;
+      margin: 0;
+      font-family: sans-serif;
+    }
+
+    .posts-list {
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      width: 100%;
+      margin: 30px auto;
+    }
+
+    .posts-list a {
+      color: black;
+      width: 85%;
+      padding: 20px;
+      margin: 30px auto;
+      border: 1px solid #ccc;
+      border-radius: 5px;
+      text-decoration: none;
+    }
+
+    .posts-list h3 {
+      margin-block: 10px;
+    }
+
+    .posts-list .user-count {
+      position: relative;
+      text-align: end;
+      justify-self: flex-end;
+      margin-block: 10px;
+    }
+
+    .posts-list ul {
+      color: black;
+      width: 85%;
+      padding: 20px;
+      margin-top: -16px;
+      margin-bottom: 20px;
+      margin-inline: auto;
+    }
+
+    .post-title {
+      display: flex;
+      flex-direction: row;
+      justify-content: space-between;
+      margin-block: 25px;
+    }
+
+    .post-details {
+      display: flex;
+      flex-direction: column;
+      margin-inline: 5%;
+    }
+
+    .post-comments {
+      display: flex;
+      flex-direction: column;
+      margin: 20px 5%;
+    }
+  </style>
+</head>
+<body>
+  <div class="posts-list">
+    <% @posts.each do |post| %>
+      <a href="<%= post_path(post) %>">
+        <div class="post-title">
+          <h3><%= post.title %></h3>
+          <div class="user-count">
+            <span>Comments: <%= post.comments.count %></span>
+            <span>Likes: <%= post.likes.count %></span>
+          </div>
+        </div>
+        <div class="post-details">
+          <p><%= post.content %></p>
+          <% if post.author %>
+            <p>Author: <%= post.author.name %></p>
+          <% else %>
+            <p>Author: Unknown</p>
+          <% end %>
+        </div>
+        <div class="post-comments">
+          <h3>Recent Comments:</h3>
+          <ul>
+            <% post.recent_comments.each do |comment| %>
+              <li>
+                <span>Username: <%= comment.author.name %></span>,
+                <span>Comment: <%= comment.text %></span>
+              </li>
+            <% end %>
+          </ul>

```

Comment:
- [ ] The five most recent comments were not yet displayed in the UI as expected, kindly implement it. 👍 

<img width="140" alt="Untitled" src="https://github.com/TayyabKM/blog-app/assets/103386183/4241dc01-9c35-4219-b570-d67a80f4546f">